### PR TITLE
[Frontend] Refactor dropzone to handle various file formats & fix display issues on Firefox

### PR DESCRIPTION
**Pull request by williammfu**
## Type of changes

 

Please check the type of change your PR introduces:
- [x] Bugfix
- [ ] Enhancement
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behaviour?

- The Dropzone component to upload photo could only accept jpeg format with file size < 800kB
- The Dropzone format display is not consistent on Mozilla Firefox

## What is the new behavior?

- Refactor DropzoneOptions component to handle various file format and size 
- Fixed DropzoneOptions css on firefox
- Screenshots (on Firefox):
## Checklist

- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have filled the `[Unreleased]` [Changelog](https://github.com/nodefluxio/weber/blob/main/CHANGELOG.md)

## Security Pentester API Checklist
- [ ] IDOR
- [ ] Malicious Script Injection
- [ ] SQL Injection

## Does this introduce a breaking change?

- [x] Yes
- [ ] No

## Other information
Kindly recheck any unhandled state that I might have missed, thanks!

### CODE REVIEW

File: frontend/app/components/elements/WarningDiv/WarningDiv.module.scss [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  * + * {
+    margin-left: 0.75rem;
+  }
+  @include phone {
+    width: 80%;
+  }
+}
+
+.warningDivWrapperActive {
+  border: 1px solid $primary-color;
```
Comment (rizkybaihaqy) [inline frontend/app/components/elements/WarningDiv/WarningDiv.module.scss:20]: kyknya kalo warning warna merah aja deh fu

File: frontend/pages/innovations/ocr-receipt.tsx [hunk 2]
```diff
@@ -15,11 +16,12 @@ const OCRReceipt = ({
   slug
 }: ServiceBySlugResponseData) => {
   const [res, setRes] = useState()
+  const MAX_IMAGE_SIZE = 3000000 // 3MB
   const renderResult = () => {
     // Error message passed
     if (typeof res !== 'undefined') {
       if (typeof res === 'string') {
-        return <div>{res}</div>
+        return <WarningDiv message={res} />
```
Comment (rizkybaihaqy) [inline frontend/pages/innovations/ocr-receipt.tsx:24]: dipersingkat pake arrow aja kali fu `=>`

File: frontend/app/components/modules/DropzoneOptions/DropzoneOptions.tsx [hunk 3]
```diff
@@ -93,7 +101,14 @@ export const DropzoneOptions = ({ images, onPhotoDrop }: Props) => {
 
   return (
     <div className={styles.dropzoneOptions}>
-      <p className={styles.errorMsg}>{errorMsg}</p>
+      <div
+        style={{
```
Comment (rizkybaihaqy) [inline frontend/app/components/modules/DropzoneOptions/DropzoneOptions.tsx:105]: gimana kalo ini dibuat kelas, jadi nanti compoenntya terima props className

File: frontend/app/components/modules/DropzoneOptions/DropzoneOptions.tsx [hunk 4]
```diff
@@ -2,27 +2,38 @@ import { MouseEvent, useCallback, useEffect, useState } from 'react'
...
+export const DropzoneOptions = ({
+  images,
+  onPhotoDrop,
+  maxSize,
+  acceptedFileFormat
+}: Props) => {
   const [photos, setPhotos] = useState<any[]>([])
   const [errorMsg, setErrorMsg] = useState<string>('')
+  const parsedFileFormat = acceptedFileFormat.split(/[\/,]+/).filter((v, i) => {
+    return i % 2 !== 0
```
Comment (rizkybaihaqy) [inline frontend/app/components/modules/DropzoneOptions/DropzoneOptions.tsx:23]: knp ga sekalian aja callback jadi `() => 1`? tujuannya ini utnuk ngambil nama filenya kan fu?

File: frontend/app/components/templates/FaceOcclusionAttributePage/FaceOcclusionAttributePage.tsx [hunk 5]
```diff
@@ -95,7 +95,9 @@ export const FaceOcclusionAttributePage = ({
       handlePost={(session_id, photo) =>
         postFaceOcclusionAttribute(id, session_id, photo)
       }
-      handleResult={(res) => setResult(res)}>
+      handleResult={(res) => setResult(res)}
+      maxImageSize={800000}
+      acceptedFileFormat={'image/jpeg'}>
```
Comment (gregoriusjimmy) [inline frontend/app/components/templates/FaceOcclusionAttributePage/FaceOcclusionAttributePage.tsx:100]: fu ini kenapa ga lanngsung jpeg aja jadi gapake image/?

File: frontend/app/components/modules/DropzoneOptions/DropzoneOptions.tsx (hunk 4 above)
Comment (rizkybaihaqy) [reply frontend/app/components/modules/DropzoneOptions/DropzoneOptions.tsx:23]: atau langsung aja gausah pake filter, `acceptedFileFormat.split(/[\/,]+/).[1]

File: frontend/app/components/elements/WarningDiv/WarningDiv.module.scss [hunk 6]
```diff
@@ -0,0 +1,22 @@
+@import '../../../styles/abstracts/variables';
+@import '../../../styles/abstracts/mixins';
+
+.warningDivWrapper {
+  color: $primary-color;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  padding: 0.3rem 1rem;
+  margin: 0.5rem auto;
```
Comment (rizkybaihaqy) [inline frontend/app/components/elements/WarningDiv/WarningDiv.module.scss:10]: menurut gw margin untuk container jangan kita set di compoennt level style de, kasi props className aja biar parentnya yang set margin

File: frontend/app/components/modules/DropzoneOptions/DropzoneOptions.module.scss [hunk 7]
```diff
@@ -9,6 +9,8 @@
   .imagePreview {
     width: inherit;
     height: inherit;
+    margin-top: 0 !important;
```
[the remark this answers is not in the document]
Comment (williammfu) [reply frontend/app/components/modules/DropzoneOptions/DropzoneOptions.module.scss:12]: Buat semua browser ky, ini efek samping pake lobotomized owl `* + *` buat gantiin `flex-gap`.
Kebetulan paling keliatan ngebug-nya di Firefox

File: frontend/pages/innovations/ocr-receipt.tsx (hunk 2 above)
Comment (williammfu) [reply frontend/pages/innovations/ocr-receipt.tsx:24]: Kalo pake fat arrow `=>` gimana tuh ky? Masih belum paham

File: frontend/app/components/modules/DropzoneOptions/DropzoneOptions.tsx (hunk 3 above)
Comment (williammfu) [reply frontend/app/components/modules/DropzoneOptions/DropzoneOptions.tsx:105]: Ohiya bener, thanks ky

File: frontend/app/components/modules/DropzoneOptions/DropzoneOptions.tsx (hunk 4 above)
Comment (williammfu) [reply frontend/app/components/modules/DropzoneOptions/DropzoneOptions.tsx:23]: Ini buat ngehandle kalo formatnya lebih dari satu ky, kea `'image/jpeg, image/png'`

File: frontend/app/components/templates/FaceOcclusionAttributePage/FaceOcclusionAttributePage.tsx (hunk 5 above)
Comment (williammfu) [reply frontend/app/components/templates/FaceOcclusionAttributePage/FaceOcclusionAttributePage.tsx:100]: Ini memang dari dokumentasi react-dropzone nya jim (referensi: https://react-dropzone.js.org/#section-accepting-specific-file-types)

File: frontend/app/components/elements/WarningDiv/WarningDiv.module.scss (hunk 6 above)
Comment (williammfu) [reply frontend/app/components/elements/WarningDiv/WarningDiv.module.scss:10]: Noted ky

File: frontend/app/components/elements/WarningDiv/WarningDiv.module.scss (hunk 1 above)
Comment (williammfu) [reply frontend/app/components/elements/WarningDiv/WarningDiv.module.scss:20]: Noted ky

File: frontend/app/components/elements/WarningDiv/WarningDiv.module.scss (hunk 6 above)
Comment (williammfu) [reply frontend/app/components/elements/WarningDiv/WarningDiv.module.scss]: Resolved on 85dfdad

File: frontend/app/components/elements/WarningDiv/WarningDiv.module.scss (hunk 1 above)
Comment (williammfu) [reply frontend/app/components/elements/WarningDiv/WarningDiv.module.scss]: Resolved on 85dfdad

File: frontend/app/components/modules/DropzoneOptions/DropzoneOptions.tsx (hunk 3 above)
Comment (williammfu) [reply frontend/app/components/modules/DropzoneOptions/DropzoneOptions.tsx]: resolved on 24942ce

**Comment by williammfu** [comment]
## Update 
**New warning display**

File: frontend/app/components/modules/DropzoneOptions/DropzoneOptions.tsx (hunk 4 above)
Comment (gregoriusjimmy) [reply frontend/app/components/modules/DropzoneOptions/DropzoneOptions.tsx:23]: owh bisa multiple format ya

File: frontend/app/components/templates/FaceOcclusionAttributePage/FaceOcclusionAttributePage.tsx (hunk 5 above)
Comment (gregoriusjimmy) [reply frontend/app/components/templates/FaceOcclusionAttributePage/FaceOcclusionAttributePage.tsx:100]: terus nanti kalo lebih dari satu input acceptedFIleFormatnya jadi array atau gimana fu?

Comment (williammfu) [reply frontend/app/components/templates/FaceOcclusionAttributePage/FaceOcclusionAttributePage.tsx:100]: Tetep satu string gitu di split pake koma (`'image/jpeg, image/png'`) gitu

File: frontend/pages/innovations/ocr-receipt.tsx (hunk 2 above)
Comment (williammfu) [reply frontend/pages/innovations/ocr-receipt.tsx:24]: if-else ada banyak, kalo pake ternary takutnya ga readable

**Review by rizkybaihaqy — APPROVED** [review]

**Review by gregoriusjimmy — APPROVED** [review]

**Pull request merged**